packages/wdio-sync/src/index.ts

```diff
@@ -5,17 +5,35 @@ import runFnInFiberContext from './runFnInFiberContext'
 import wrapCommand from './wrapCommand'
 
 import { STACKTRACE_FILTER_FN } from './constants'
-const defaultRetries = { attempts: 0, limit: 0 }
+
+declare global {
+  namespace NodeJS {
+      interface Global {
+          _HAS_FIBER_CONTEXT: boolean
+          WDIO_WORKER: boolean
+          browser: {
+              _NOT_FIBER?: boolean
+          }
+      }
+  }
+}
+
+export type Retries = {
+    attempts: number;
+    limit: number;
+}
+
+const defaultRetries: Retries = { attempts: 0, limit: 0 }
 
 /**
  * execute test or hook synchronously
  *
  * @param  {Function} fn         spec or hook method
- * @param  {Number}   retries    { limit: number, attempts: number }
+ * @param  {Retries}   retries    { limit: number, attempts: number }
  * @param  {Array}    args       arguments passed to hook
  * @return {Promise}             that gets resolved once test/hook is done or was retried enough
  */
-async function executeSync(fn, retries = defaultRetries, args = []) {
+async function executeSync<T>(this: any, fn: Function, retries = defaultRetries, args: unknown[] = []): Promise<T> {
```

Comment:
```suggestion
async function executeSync<T>(this: any, fn: Function, retries = { attempts: 0, limit: 0 }, args: unknown[] = []): Promise<T> {
```

Reply:
The problem is now with your suggestion, we have duplication, since in `runSync` there is: `repeatTest = defaultRetries`, which references `defaultRetries`, which is `{ attempts: 0, limit: 0 }` as well..